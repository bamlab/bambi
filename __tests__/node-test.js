const eslint = require('eslint');
const path = require('path');

const checkPrettierRulesAsync = require('./tools/checkPrettierRulesAsync');
const lintAsync = require('./tools/lintAsync');

const configFile = path.resolve(__dirname, '../node.js');

it(`has a Node config`, () => {
  expect(
    () =>
      new eslint.CLIEngine({
        configFile,
        useEslintrc: false,
      })
  ).not.toThrow();
});

it(`lints with the Node config`, async () => {
  let report = await lintAsync(
    {
      configFile,
      fix: true,
      ignore: false,
      useEslintrc: false,
    },
    ['__tests__/fixtures/*all*.js', '__tests__/fixtures/*node*.js']
  );
  let { results } = report;
  for (let result of results) {
    let relativeFilePath = path.relative(__dirname, result.filePath);
    delete result.filePath;
    expect(result).toMatchSnapshot(relativeFilePath);
  }
});

it(`doesn't conflict with Prettier`, async () => {
  let { success, message } = await checkPrettierRulesAsync(configFile);
  expect(success).toMatchSnapshot('success');
  expect(message).toMatchSnapshot('message');
});
