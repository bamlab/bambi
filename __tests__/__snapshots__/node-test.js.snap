// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures/all-00.js 1`] = `
Object {
  "errorCount": 0,
  "fixableErrorCount": 0,
  "fixableWarningCount": 0,
  "messages": Array [],
  "output": "import 'hi';

import a from 'a';
import { b } from 'b';
import c, { c1 } from 'c';
import * as d from 'd';

import e from './e';

@d
export class Example {
  static s = {
    a,
    b,
  };

  m = {};

  hi() {
    _fn();
  }

  async byeAsync() {
    _fn(a, b, c, c1, d, e);
  }
}

function _fn() {}
",
  "warningCount": 0,
}
`;

exports[`message 1`] = `
"No rules that are unnecessary or conflict with Prettier were found.
"
`;

exports[`success 1`] = `true`;
